packages/maskbook/src/plugins/RedPacket/SNSAdaptor/hooks/useNftRedPacketHistory.ts

```diff
@@ -0,0 +1,10 @@
+import { useAsyncRetry } from 'react-use'
+import { RedPacketRPC } from '../../messages'
+import type { ChainId } from '@masknet/web3-shared'
+
+export function useNftRedPacketHistory(address: string, chainId: ChainId) {
+    return useAsyncRetry(async () => {
+        const payloads = await RedPacketRPC.getNftRedPacketHistory(address, chainId)
```

Comment:
It would be nice to have a lazy-loading here.

[ITO/SNSAdaptor/hooks/useAllPoolsAsSeller.ts](https://github.com/DimensionDev/Maskbook/blob/develop/packages/maskbook/src/plugins/ITO/SNSAdaptor/hooks/useAllPoolsAsSeller.ts)

[ITO/SNSAdaptor/PoolList.tsx#L39](https://github.com/DimensionDev/Maskbook/blob/develop/packages/maskbook/src/plugins/ITO/SNSAdaptor/PoolList.tsx#L39)

Reply:
Added pagination.